subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/result/DefaultResolvedComponentResult.java

```diff
@@ -152,16 +152,25 @@ public void associateDependencyToVariant(DependencyResult dependencyResult, Reso
         variantDependencies.put(fromVariant, dependencyResult);
     }
 
+    /**
+     * A recursive function that traverses the dependency graph of a given module and acts on each node and edge encountered.
+     *
+     * @param start A ResolvedComponentResult node, which represents the root of the dependency graph to be traversed
```

Comment:
I think this is only true for the first call to `eachElement`. Since this is recursive, after subsequent calls, `start` is no longer the root of the graph